src/Core/PluginCollection.php

```diff
@@ -285,8 +285,8 @@ public function create(string $name, array $config = []): PluginInterface
                 throw new InvalidArgumentException(sprintf('Class `%s` does not exist.', $name));
             }
 
-            /** @var \Cake\Core\PluginInterface $plugin */
             $plugin = new $name($config);
+            assert($plugin instanceof PluginInterface);
```

Comment:
I don't like such asserts just to please the static analyzers. All we are doing here is potentially changing a `TypeError` to `AssertError` which from a user POV makes no practical difference.

Reply:
That's fair. I'll revert these kind of changes.